corehq/apps/userreports/laboratory/adapter.py

```diff
@@ -56,3 +56,10 @@ def best_effort_save(self, doc, eval_context=None):
     def _save_rows(self, rows, doc):
         self.es_adapter._save_rows(rows, doc)
         self.sql_adapter._save_rows(rows, doc)
+
+    def delete(self, doc):
+        self.es_adapter.delete(doc)
+        self.sql_adapter.delete(doc)
+
+    def doc_exists(self, doc):
+        return self.es_adapter.doc_exists(doc) or self.sql_adapter.doc_exists(doc)
```

Comment:
@emord Only one will throw the error. I'm checking now if doc exist before deleting it.